host/http.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/host/types"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/shutdown"
+	"github.com/flynn/flynn/pkg/sse"
+)
+
+func serveHTTP(host *Host, attach *attachHandler, sh *shutdown.Handler) (*httprouter.Router, error) {
+	l, err := net.Listen("tcp", ":1113")
+	if err != nil {
+		return nil, err
+	}
+	sh.BeforeExit(func() { l.Close() })
+
+	r := httprouter.New()
+	r.POST("/attach", attach.ServeHTTP)
+	r.GET("/host/jobs", hostMiddleware(host, listJobs))
+	r.GET("/host/jobs/:id", hostMiddleware(host, getJob))
+	r.DELETE("/host/jobs/:id", hostMiddleware(host, stopJob))
```

Comment:
Instead of `hostMiddleware`, make a struct:

```go
type httpAPI struct {
  Host *Host
}
```

And then just define methods on it and pass method values to the router.